src/main/java/org/trigon/hardware/rev/spark/SparkMotor.java

```diff
@@ -1,134 +1,184 @@
 package org.trigon.hardware.rev.spark;
 
-import com.revrobotics.CANSparkBase;
-import com.revrobotics.CANSparkLowLevel;
-import com.revrobotics.SparkPIDController;
+import com.revrobotics.spark.SparkBase;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.config.SparkBaseConfig;
 import edu.wpi.first.math.system.plant.DCMotor;
-import edu.wpi.first.wpilibj.Timer;
 import org.littletonrobotics.junction.Logger;
 import org.trigon.hardware.RobotHardwareStats;
 import org.trigon.hardware.rev.spark.io.RealSparkIO;
 import org.trigon.hardware.rev.spark.io.SimulationSparkIO;
+import org.trigon.hardware.simulation.MotorPhysicsSimulation;
 
-import java.util.concurrent.CompletableFuture;
-
+/**
+ * A class the represents a Spark motor. Used to control and read data from a Spark motor.
+ */
 public class SparkMotor {
     private final String motorName;
     private final SparkIO motorIO;
     private final SparkInputs motorInputs;
     private final int id;
 
-    public SparkMotor(int id, SparkType sparkType, String motorName, DCMotor simulationMotor) {
+    /**
+     * Creates a new Spark motor.
+     *
+     * @param id        the motor's ID
+     * @param sparkType the type of Spark motor
+     * @param motorName the name of the motor
+     */
+    public SparkMotor(int id, SparkType sparkType, String motorName) {
         this.id = id;
         this.motorName = motorName;
         motorInputs = new SparkInputs(motorName);
-        motorIO = createSparkIO(id, sparkType, simulationMotor);
+        motorIO = createSparkIO(id, sparkType);
     }
 
+    /**
+     * Processes the inputs of the motor.
+     */
     public void update() {
         Logger.processInputs("Motors/" + motorName, motorInputs);
+        motorIO.updateSimulation();
     }
 
     public int getID() {
         return id;
     }
 
-    public void registerSignal(SparkSignal signal) {
-        this.registerSignal(signal, false);
+    /**
+     * Registers a threaded signal to be logged from the motor.
+     * Threaded signals use threading to process certain signals separately. Should be used for signals that need to be updated at a high frequency such as odometry.
```

Comment:
Threaded signals use threading to process certain signals separately at a faster rate.
Keep the jdoc consistent across all threaded stuff